tests/unit/s2n_ktls_test.c

```diff
@@ -327,70 +384,111 @@ int main(int argc, char **argv)
             server_conn->managed_recv_io = true;
             EXPECT_SUCCESS(s2n_connection_ktls_enable_recv(server_conn));
         };
-    };
 
-    /* Test s2n_ktls_init_aes128_gcm_crypto_info */
-    {
-        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
-                s2n_connection_ptr_free);
-        EXPECT_OK(s2n_test_configure_connection_for_ktls(server_conn));
-
-        /* prepare test data */
-        uint8_t test_data[S2N_MAX_KEY_BLOCK_LEN] = { 0 };
-        struct s2n_blob test_data_blob = { 0 };
-        EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
-        EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
-
-        /* copy test data to key_material */
-        struct s2n_key_material key_material = { 0 };
-        EXPECT_OK(s2n_key_material_init(&key_material, server_conn));
-        POSIX_CHECKED_MEMCPY(key_material.key_block, test_data, s2n_array_len(key_material.key_block));
-
-#if defined(S2N_KTLS_SUPPORTED)
-        struct tls12_crypto_info_aes_gcm_128 crypto_info = { 0 };
-
-        /* generate test data for crypto params: implicit_iv and sequence_number */
-        EXPECT_OK(s2n_test_generate_fake_crypto_params(server_conn));
-        struct s2n_crypto_parameters *server_param = server_conn->server;
-
-        /* Test crypto_info for happy case */
+        /* Call setsockopt correctly to configure tls crypto */
```

Comment:
I think I'm understanding the test now, but I'm not sure I understand this name. Since setsockopt and set_ktls_info are overridden to just check the values in the `s2n_test_setsockopt_expected` struct, isn't this test just making sure that `s2n_ktls_crypto_info_init()` creates the correct crypto info, which is correctly passed to `s2n_setsockopt()` in `s2n_connection_ktls_enable()`? It's not testing anything about setsockopt itself right?

Is this name accurate?
```suggestion
        /* setsockopt is called with correctly configured ktls crypto info */
```

Reply:
I'm not sure I understand how the original name implies we're testing setsockopt itself or how the old name is different from the new name, but I can use the new name if you prefer it.